code/modules/ai/ai_behaviors/human_mobs/grenades.dm

```diff
@@ -0,0 +1,43 @@
+/datum/ai_behavior/human
+	///Probability of trying to throw a grenade during process
+	var/nade_throw_prop = 15
+	///Chat lines for throwing a nade
+	var/list/nade_throw_chat = list("Grenade out!", "Fire in the hole!", "Grenade!", "Catch this!")
+
+///Decides if we should throw a grenade
+/datum/ai_behavior/human/proc/grenade_process()
+	if(!combat_target)
+		return
+	if(!prob(nade_throw_prop))
+		return
+	if(!length(mob_inventory.grenade_list))
+		return
```

Comment:
may want to toggle something when we run out of/gain a new grenade instead

Reply:
True. Or I could just do this check first? Since AI is processing either way, it has to check something to not run the rest of the code, so this should be fine